src/searchmultipleEndpoint.js

```diff
@@ -0,0 +1,149 @@
+/*
+    ***** BEGIN LICENSE BLOCK *****
+
+    Copyright © 2018 Corporation for Digital Scholarship
+                     Vienna, Virginia, USA
+                     https://www.zotero.org
+
+    This file is part of Zotero.
+
+    Zotero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Zotero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+
+    ***** END LICENSE BLOCK *****
+*/
+
+const config = require('config');
+const Translate = require('./translation/translate');
+const TextSearch = require('./textSearch');
+
+const SELECT_TIMEOUT = 60;
+const sessionsWaitingForSelection = {};
+
+var requestsSinceGC = 0;
+
+var SearchMultipleEndpoint = module.exports = {
+	handle: async function (ctx, next) {
+		ctx.assert(ctx.is('text/plain') || ctx.is('json'), 415);
+
+        setTimeout(() => {
+            gc();
+        });
+
+
+		var data = ctx.request.body;
+
+		if (!data) {
+			ctx.throw(400, "POST data not provided\n");
+		}
+
+        // If follow-up request, retrieve session and update context
+		var query;
+		var session;
+		if (typeof data == 'object') {
+			let sessionID = data.session;
+			if (!sessionID) {
+				ctx.throw(400, "'session' not provided");
+			}
+			session = sessionsWaitingForSelection[sessionID];
+			if (session) {
+				delete sessionsWaitingForSelection[sessionID];
+				session.ctx = ctx;
+				session.next = next;
+				session.data = data;
+			} else {
+				let single = !!ctx.request.query.single;
+				session = new SearchMultipleSession(ctx, next, data, { single });
+			}
+		}
+		else {
+            // Look for DOI, ISBN, etc.
```

Comment:
These two lines seem to be whitespace-indented, rest is tab-indented

Reply:
... seems to be a problem in other places too. can you please normalize the file?